### PR TITLE
Addition of DataUpload Script

### PR DESCRIPTION
**Pull request by Joel-Joseph-George**
#724

### CODE REVIEW

File: Dataupload/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+certifi==2021.10.8
+charset-normalizer==2.0.12
+idna==3.3
+psycopg2
+requests==2.27.1
+urllib3==1.26.8
+fastapi[all]==0.95.0 
+SQLAlchemy==2.0.9
+jsonpickle==2.2.0
+pytz==2023.3
```
Comment (shimilgithub) [inline Dataupload/requirements.txt]: does all the packages are used here. especially such as jsonpickle?

File: Dataupload/upload.py [hunk 2]
```diff
@@ -0,0 +1,644 @@
...
+def create_database_schema():
+    postgres_host = os.environ.get("VACHAN_POSTGRES_HOST", "localhost")
+    postgres_user = os.environ.get("VACHAN_POSTGRES_USER", "postgres")
+    postgres_database = os.environ.get("VACHAN_POSTGRES_DATABASE", "vachan_db")
+    postgres_password = os.environ.get("VACHAN_POSTGRES_PASSWORD", "secret")
+    postgres_port = os.environ.get("VACHAN_POSTGRES_PORT", "5432")
+    postgres_schema = os.environ.get("VACHAN_POSTGRES_SCHEMA", "vachan_cms_rest_12")
+
+    #if you want to use a new schema, you can use below code to specify the name.
+    # postgres_schema = os.environ.get("VACHAN_POSTGRES_SCHEMA", "DataUpload")
```
Comment (shimilgithub) [inline Dataupload/upload.py]: change `DataUpload` to ` "<schema_name>"`

File: Dataupload/upload.py [hunk 3]
```diff
@@ -0,0 +1,644 @@
...
+    if postgres_schema in existing_schemas:
+        print(f"Schema '{postgres_schema}' already exists.")
+    else:
+        conn.execute(CreateSchema(postgres_schema))
+        conn.commit()
+        print(f"Schema '{postgres_schema}' created successfully.")
+
+
+#creating token
+        #Token is now disabled since cms is not integrated with auth. When it is , the token need to be enabled with headers
```
Comment (shimilgithub) [inline Dataupload/upload.py]: remove first sentence

File: Dataupload/upload.py [hunk 4]
```diff
@@ -0,0 +1,644 @@
...
+                                "chapterEnd":fields[1],
+                                "verseEnd": fields[3]},
+
+                    'commentary': fields[4],
+                    'active': fields[5] == 't',
+                    "sectionType":["commentary-text"]
+                })
+    # for item in parsed_data:
+    #     print(item)
+    # print(parsed_data)
```
Comment (shimilgithub) [inline Dataupload/upload.py]: remove if not required

File: Dataupload/upload.py [hunk 5]
```diff
@@ -0,0 +1,644 @@
...
+        print(f"Error: File '{csv_file_path}' not found.")
+    except Exception as e:
+        print(f"An error occurred while processing {csv_file_path}: {str(e)}")
+
+    print("data,", usfm_list)
+
+    return usfm_list
+
+#==========================================================================================================================
+# def add_parascriptual(csv_file_path):   #Only use if you want to add new parascriptual data using folder "parascriptuals"
```
Comment (shimilgithub) [inline Dataupload/upload.py]: enable this  function. Just mention initially like this  to enable function if required. But functions should not be commented.

File: Dataupload/upload.py [hunk 6]
```diff
@@ -0,0 +1,644 @@
...
+                    except Exception as e:
+                        print(f"Failed to upload data for {resource_name}: {str(e)}")
+
+                #By this method you can validate file by file
+                # try:
+                # # Upload all entries for a single CSV file
+                #     upload_data(data, bible_url)
+                #     print(f"Success: All data for {resource_name} uploaded successfully.")
+                # except Exception as e:
+                #     print(f"Failed to upload data for {resource_name}: {str(e)}")
```
Comment (shimilgithub) [inline Dataupload/upload.py]: remove if not required. Else uncomment it

File: Dataupload/upload.py [hunk 7]
```diff
@@ -0,0 +1,644 @@
...
+
+    data = add_audiobible_data()
+    resource_name = 'dgo_DSV_1_audiobible'
+    audiobible_url = f"/resources/bible/audios/{resource_name}"
+    upload_data(data, audiobible_url)
+
+    #8th
+    #Add commentaries
+
+    file_paths = [
```
Comment (shimilgithub) [inline Dataupload/upload.py:616]: add a comment that these are the data files

File: Dataupload/fetching.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
+import os
```
Comment (shimilgithub) [inline Dataupload/fetching.py]: give a docstring  to mention what is there in this file

Comment (shimilgithub) [reply Dataupload/fetching.py]: find a better name for this file.

File: Dataupload/readme.MD [hunk 9]
```diff
@@ -0,0 +1,50 @@
+# DataUpload
```
Comment (shimilgithub) [inline Dataupload/readme.MD]: put a space. "Data Upload"

**Review by shimilgithub — CHANGES_REQUESTED** [review]

File: Dataupload/requirements.txt (hunk 1 above)
Comment (Joel-Joseph-George) [reply Dataupload/requirements.txt]: sorry bout that, i kept the previous requirement file as it is.. i will send the updated one soon

**Review by shimilgithub — APPROVED** [review]

**Pull request merged**